Styles for [input:disabled] on checkboxes and radio. Issue #17918

markau opened the pull request:
Fixes #17918.

markau committed:
Styles for [input:disabled] on checkboxes and radio. Issue #17918

markau committed:
Styles for [input:disabled] on checkboxes and radio elements. Issue #17918

cvrebert commented:
If possible, exclude `dist/css/bootstrap.css*` and `docs.min.css*` from the PR.

cvrebert commented:
Also, it's not required, but you might want to associate the email address you used for your commit with your GitHub account so that you'll get credit for it in your GitHub profile page.

markau commented:
Thanks for the Git help, @cvrebert, much appreciated.  I have added the email to my GitHub profile (and updated git config with the preferred one so this doesn't happen again).

I tried to `git rm` the files, then commit with the `--ammend` flag to correct the PR and remove those files. I think I just made a mess of my local repo though. Should I try some more?  (also, would a push following an ammended commit update the email address used for the commit?)

I'd like to get this right, but don't want to waste your time.

Thank you!

markau commented:
Ok I realized the commits had not been pushed. I *think* the commit is now correct (i.e. rm'd the dist\css files). Interested in how it looks to you, @cvrebert.  Thank you!

cvrebert commented:
> Ok I realized the commits had not been pushed. I think the commit is now correct (i.e. rm'd the dist\css files). Interested in how it looks to you

Doesn't quite look right. (The files I mentioned shouldn't be in the diff at all.)

> Should I try some more?

Nah, that's all right. We can fix it on our end when merging this.
(We'll use "git rebase" to do the fixing; you can google around for info on rebasing if you're interested, but it's somewhat advanced.)

> (also, would a push following an amended commit update the email address used for the commit?)

To update the author info, you'd need to do `git commit --amend --reset-author` , and then do `git push --force` (which should be used with caution since it can "delete" (I'm oversimplifying) commits, but should be safe in this case).

markau commented:
Thanks for the info, Chris.  I will have a go at updating the email tomorrow when back in the office.

markau committed:
Styles for [input:disabled] on checkboxes and radio. Issue #17918

cvrebert commented on scss/_custom-forms.scss:
```diff
@@ -83,6 +83,10 @@
   }
+
+  input:disabled ~ .c-indicator {
+    background-color: #bbb;
```
Should probably use a variable for this color

markau commented:
I did wonder that and I'll take a look. Thanks for your patience, Chris. I have used open source software for years and this is my first meaningful contribution to a project, so I am happy to make the effort to get it right. 

If I make a change to use a variable, would it be a new commit/push (referencing this issue number?)

cvrebert commented:
No problem! :smile: Thanks for giving back to the community :heart: 

The variable declaration should be added to [`_variables.scss`](https://github.com/twbs/bootstrap/blob/v4-dev/scss/_variables.scss)

> If I make a change to use a variable, would it be a new commit/push (referencing this issue number?)

Yes, that'd be fine.

markau commented:
I'll include the doc changes in the commit, but that is probably the most subjective part because you might not like the wording. Ideally, I would show you a diff of the changes for review prior to committing. How best to achieve this?

cvrebert commented:
Eh, you might as well just commit your suggestion and then depending on the feedback either you can push more fixup commits or we can fix it ourselves when we go to merge it. In any case, we'll squash it into a single commit on our end when we merge this PR.

markau committed:
Using a variable for the disabled element color. Issue #17918

markau commented:
Thanks once again, Chris. I'll add the doco commit later when I can test (Jekyll setup is problematic on this machine.)

Commit:
Color variables for both states. Improved colours. Issue #17918

markau committed:
Debugging removed. Issue #17918

markau committed:
Label color now reflects element state. Issue #17918

markau committed:
Adding cursor:not-allowed to element and label. Issue #17918

markau committed:
Keeping the hound at bay. Fixed formatting. Issue #17918

markau committed:
Formatting for hound. Issue #17918

markau commented:
Hello Chris, me again. While I have tested every change prior to commit, I clearly haven't linted. Sorry for the hound spam. Lesson learned.

Can I just check whether you are likely to be happy with wrapping the checkbox/radio label in a span? This is in order to apply the `cursor: none;` style. I ask because it adds a relatively large chunk of characters to the html (every time an element is used) for what is a small detail (and only pertinent when the element is disabled). i.e. 

before:

    <label class="c-input c-checkbox">
        <input type="checkbox" checked="checked">
        <span class="c-indicator"></span>
        Can toggle this
    </label>

after:

    <label class="c-input c-checkbox">
        <input type="checkbox" checked="checked">
        <span class="c-indicator"></span>
        <span class="c-indicator-label">Can toggle this</span>
    </label>

I tend to think the cleaner the html the better, but I see the value either way. I'm curious which way you'd want to go.

cvrebert commented:
> Can I just check whether you are likely to be happy with wrapping the checkbox/radio label in a span? 

Those sorts of design specifics regarding the custom form widgets are @mdo's department rather than mine.

markau committed:
Formatting - trailing whitespace. #17918

mdo commented:
Changes from here are now part of #18771 (squashed and rebased there). Thanks!

markau commented:
Thanks @mdo, it's great to have made even a small contribution. Will I be famous and show up as a contributor? (I'm a simple man :blush:)

mdo commented:
Hah, yeah, you'll be on the contributors list once the `v4-dev` branch makes it into `master` :).
